examples/flow.cpp

```diff
@@ -116,18 +76,31 @@ namespace detail
 int main(int argc, char** argv)
 {
     // MPI setup.
-    // Must ensure an instance of the helper is created to initialise MPI.
-    // For a build without MPI the Dune::FakeMPIHelper is used, so rank will
-    // be 0 and size 1.
-    const Dune::MPIHelper& mpi_helper = Dune::MPIHelper::instance(argc, argv);
-    const bool outputCout = mpi_helper.rank() == 0;
+#if HAVE_DUNE_FEM
+    Dune::Fem::MPIManager::initialize(argc, argv);
+    int mpiRank = Dune::Fem::MPIManager::rank();
+#else
+    // the design of the plain dune MPIHelper class is quite retarted: there is no way to
```

Comment:
Please adjust language tone...

Reply:
s/retarted/flawed/ (although I stick to my original formulation -- this API just begs to be used incorrectly)